### PR TITLE
创建：简单加密

### PR DESCRIPTION
**Pull request by Starry-PCND**
版本V1.0
### **拓展文档**
https://www.ccw.site/post/a810ef3e-edfc-49ba-a70d-e10e54bf11c6
积木图片
c11f7a44c1c9e68bd14ec7e021f5a763

### CODE REVIEW

**Comment by Starry-PCND** [comment]
When will the approver go live?

**Comment by MoreBugOfDog** [comment]
Personally, I don't think the crypto class extension is needed anymore, since there are arkos related extensions. (Of course, you can keep this PR)

**Comment by MoreBugOfDog** [comment]
You could try “Extension Lab Square.”（Chinese:拓展实验广场）

File: extensions/Starry/SimpleEncrypt.js [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
...yhq8owTvbC7RQ9saNF9G6jjwDl8coZxg872yjnjdNeadodx896hFi2chdhBwQ6J+CaHXADHsXapB3gstt0cOKMXuAiyKyC+gexv9yEYN0pBbhdfc7hGkM+m73TCU0ywOrhpZMAnlX3eNy9N4BMNnrLah9a9hDOU7Zb2UymX095HMW8RBg8zTgKO6zpqwMhVCVLA57lOViZbek8xOfamAbarzTFmmRvc5hiIEHgjexHMi9pia6imTplTGNHfSu0CIWIkjq8y1zfmNgdtDjfM5InJbW6AGcPmyBfGJ5bxYX1/pcc9NkKmqaHSRLtCznrsbQ4iU8TJARcjSEJ/YyMxscyY0iZLuca59LX29VR91bnGzPyoIjKR66m0zaAdy5mL7ab7BZlOEXZXYtLAkciIdgt7b7DzWHd9RtlPz34eYsTWcBFi77mTALguKvhGtiePbG7RfRuJ7t/Qoie2zThtgucv64EbVhDdPdyiu9a06OolACOmUjsjB57y2E01PgEYG+WhUYnHkFslIcZRjnY9zNvKvHC4SIPjAGYn1F+QedFkjRp4HU11jRoNnpLmx3Y6+M5TGZlHEJ6SmNEafVPxlMfuInbbxjVqnGHvyKD9y4mLOvsXknGTwByy8UmHDIDMazXDx3/rU+VwIZm31NwuT26/1GThaTOLouh4rOcL7qelkO3p6/yxIrJRWRSuS9XT6eyROYW+0kfom6OKvtaiFJ/nxtde1QLIjI3guMPvfkSlSp4IQzH15t8sY0SaT4k5lHl+uPHJpq8wv2KETLEuavPiWRf2sJX7gmKctfqWBXcZUO7dv6x1DM+LG8rQtpcqX7SHRArchRXob22cPEwPN0AJjS+ir/siW19EntFX3Be49pjDg2T/KvuWlhkTsNdeBpv9vXY/rpodbBInJOCMbPZTY68lvBCYXE7fDPjmwAmiBze26P6u48bcMDXzw0Zg2Wyi9w2b+jZtunOIaNFsLi221/besx5d+BCLx1i7q2Sylk6LVHsPMJC8ODAwTpV+CvYAYsKCTBqvPQPlRfunsfeAvQ7i3AAeC46vY//c6QrKC/gEsme9xlOR9r6jZiyecs7ZKjwVo28unGE7b7C9ukwrKtctguMS7HU5LzHjJDmJJnXCatR4mTGgByw291wROZosGOTWmmE+Ql+EjHPaVPa8QH9j9c3SbrFIAMYG6/FXyHHpY
```
Comment (MoreBugOfDog) [inline extensions/Starry/SimpleEncrypt.js:261]: Regarding the readability of variable names: it is not recommended to use a mixture of Hanyu Pinyin initials + English semantics. (Recommendations only)

[the remark this answers is not in the document]
Comment (Starry-PCND) [reply extensions/Starry/SimpleEncrypt.js:275]: Good! I'll make revisions in my free time during my vacation

[the remark this answers is not in the document]
Comment (Starry-PCND) [reply extensions/Starry/SimpleEncrypt.js:275]: Good! It's a pity that I have a heavy academic load and have no time to revise it for the time being, I hope you can agree to this PR, thank you very much!

**Comment by Starry-PCND** [comment]
> Personally, I don't think the crypto class extension is needed anymore, since there are arkos related extensions. (Of course, you can keep this PR)

I appreciate your perspective, but I believe the "Simple Crypto" extension still holds its unique value. While Arkos does offer a method of encryption, the "Simple Crypto" class extension provides a variety of encryption options. Although Arkos encryption can handle non-numeric values, the "Simple Crypto" class extension can also utilize decimals and negative numbers as keys, which is an added feature. Moreover, while Arkos encryption is essentially a traditional approach, the "Simple Crypto" class extension offers more secure encryption techniques. Of course, the decision to merge the PR is ultimately up to you.

**Comment by fath11** [comment]
i recommend renaming this extension to "Simple Encryption". The word "Crypto" does not age well

**Comment by Starry-PCND** [comment]
重大更新SimpleEncrypt.js
增加加密选项：
 - RSA加密/解密
 - DES加密/解密
 - RC4加密/解密
 - MD5加密
 - AES加密/解密
 - AES加密/解密
 - 3DES加密/解密
全局增加功能：
 - 密钥字符无限制

Added encryption options:
 - RSA encryption/decryption
 - DES encryption/decryption
 - RC4 encryption/decryption
 - MD5 encryption
 - AES encryption/decryption
 - AES encryption/decryption
 - 3DES encryption/decryption
Global Additions:
 - There is no limit to the key characters

@fath11 @MoreBugOfDog

File: extensions/Starry/SimpleEncrypt.js [hunk 2]
```diff
@@ -0,0 +1,749 @@
+// Github ID: Starry-PCND [https://github.com/Starry-PCND/]
+// 共创世界 ID: Starry [https://www.ccw.site/student/641eaa83b1ef645c6d6e8e06/]
+// bilibili ID: 柚子路灯 [https://space.bilibili.com/1581040466/]
+// 代码协议: BSD 3-Clause / Code Protocol: BSD 3-Clause
```
Comment (MoreBugOfDog) [inline extensions/Starry/SimpleEncrypt.js:4]: 首先，在源代码中添加自己的信息毫无意义 这不会被引用到官方拓展市场 bilibili也不是我们交流的渠道 
代码不是你宣传自己的地方哦
其次 代码协议 英文应由 许可证 意译

File: extensions/Starry/SimpleEncrypt.js [hunk 3]
```diff
@@ -0,0 +1,749 @@
...
...b2UymX095HMW8RBg8zTgKO6zpqwMhVCVLA57lOViZbek8xOfamAbarzTFmmRvc5hiIEHgjexHMi9pia6imTplTGNHfSu0CIWIkjq8y1zfmNgdtDjfM5InJbW6AGcPmyBfGJ5bxYX1/pcc9NkKmqaHSRLtCznrsbQ4iU8TJARcjSEJ/YyMxscyY0iZLuca59LX29VR91bnGzPyoIjKR66m0zaAdy5mL7ab7BZlOEXZXYtLAkciIdgt7b7DzWHd9RtlPz34eYsTWcBFi77mTALguKvhGtiePbG7RfRuJ7t/Qoie2zThtgucv64EbVhDdPdyiu9a06OolACOmUjsjB57y2E01PgEYG+WhUYnHkFslIcZRjnY9zNvKvHC4SIPjAGYn1F+QedFkjRp4HU11jRoNnpLmx3Y6+M5TGZlHEJ6SmNEafVPxlMfuInbbxjVqnGHvyKD9y4mLOvsXknGTwByy8UmHDIDMazXDx3/rU+VwIZm31NwuT26/1GThaTOLouh4rOcL7qelkO3p6/yxIrJRWRSuS9XT6eyROYW+0kfom6OKvtaiFJ/nxtde1QLIjI3guMPvfkSlSp4IQzH15t8sY0SaT4k5lHl+uPHJpq8wv2KETLEuavPiWRf2sJX7gmKctfqWBXcZUO7dv6x1DM+LG8rQtpcqX7SHRArchRXob22cPEwPN0AJjS+ir/siW19EntFX3Be49pjDg2T/KvuWlhkTsNdeBpv9vXY/rpodbBInJOCMbPZTY68lvBCYXE7fDPjmwAmiBze26P6u48bcMDXzw0Zg2Wyi9w2b+jZtunOIaNFsLi221/besx5d+BCLx1i7q2Sylk6LVHsPMJC8ODAwTpV+CvYAYsKCTBqvPQPlRfunsfeAvQ7i3AAeC46vY//c6QrKC/gEsme9xlOR9r6jZiyecs7ZKjwVo28unGE7b7C9ukwrKtctguMS7HU5LzHjJDmJJnXCatR4mTGgByw291wROZosGOTWmmE+Ql+EjHPaVPa8QH9j9c3SbrFIAMYG6/FXyHHpYsg4scyCmRxw4uZDrPmiYbYoc5s1ERxb72CEnolIQkwTMPIkMmOAiKIUGYRyqjNGYlTrgYlEQkwcYV1A8kx/IblADRUkglMDGZyzQohUlRiCAcXIOi7tJ2j/ahEgRF/OjinlJIZLkEHEgSKwv7H2PglnOPrah1OREZ
```
Comment (MoreBugOfDog) [inline extensions/Starry/SimpleEncrypt.js:9]: 过长的dataurl可以使用m.ccw.site渠道（canvas拓展）
当然这不是必须的

Comment (MoreBugOfDog) [inline extensions/Starry/SimpleEncrypt.js:24]: 与 op七星松 的拓展重复

Comment (MoreBugOfDog) [inline extensions/Starry/SimpleEncrypt.js:69]: 摘要算法和加密不是一个概念

**Review by MoreBugOfDog — COMMENTED** [review]
我不是官方开发人员 仅做建议
此拓展我认为没有实际意义
should close

File: extensions/Starry/SimpleEncrypt.js (hunk 3 above)
Comment (kukemc) [reply extensions/Starry/SimpleEncrypt.js:9]: > 过长的dataurl可以使用m.ccw.site渠道（canvas拓展） 当然这不是必须的

这就没必要了，m.ccw.site只是ccw国内的图床 ，对于cocrea 国外用户来讲加载速度会很慢，其次是上传的图片url并不能保证长久稳定性

**Pull request closed without merging**